pkg/sql/row/deleter.go

```diff
@@ -165,13 +165,15 @@ func (rd *Deleter) DeleteRow(
 		rd.key = keys.MakeFamilyKey(primaryIndexKey, uint32(familyID))
 
 		if oth.IsSet() {
-			prevValue, err := rd.encodeValueForPrimaryIndexFamily(family, values)
-			if err != nil {
-				return err
-			}
 			var expValue []byte
-			if prevValue.IsPresent() {
-				expValue = prevValue.TagAndDataBytes()
+			if !oth.PreviousWasDeleted {
+				prevValue, err := rd.encodeValueForPrimaryIndexFamily(family, values)
```

Comment:
why is this extra logic around `oth.PreviousWasDeleted` only on the delete path, and not on the insert/update path?

Also, it's only possible for `!oth.PreviousWasDeleted && !prevValue.IsPresent()` if the prev value is null right?

Reply:
Good question, I'll write some words about this, maybe I overlooked something this morning.

Here is the issue.  A _parsed_ rangefeed event for a delete contains some of the columns. Namely, it contains any columns in the primary key.  This is also true for a _phantom delete_.  But, for a phantom delete, the correct expected value is nothing.  We need the columns from the key to construct the key, but when constructing the _expectation_ we want to make sure that we expect no key (as opposed to an existing key with a sentinel body).

Reply:
yuck. given that the write will eventually succeed on a bogus expected value, i wonder it if is even worth having this extra logic. At the very least, perhaps we document this here?